challenge/src/main/java/com/ifood/challenge/location/ExceptionController.java

```diff
@@ -0,0 +1,19 @@
+package com.ifood.challenge.location;
+
+import org.springframework.http.HttpStatus;
+import org.springframework.web.bind.annotation.ControllerAdvice;
+import org.springframework.web.bind.annotation.ResponseStatus;
+
+@ControllerAdvice
+public class ExceptionController {
+
+    @ResponseStatus(HttpStatus.BAD_REQUEST)
+    public static class BadArgumentsException extends RuntimeException {
+
+        public BadArgumentsException(String message) {
+            super(message);
+        }
+    }
+
+}
```

Comment:
A implementação do `ControllerAdvice` não é bem isso. No caso, o advice não está tratando `BAD_REQUEST`. Se `ExceptionController` for removida, os testes basicamente continuarão passando.

A implementação correta do advice envolve ter métodos anotados com `@ExceptionHandler` para exceções específicas.

Esse é um motivo adicional pelo qual adicionar o advice no projeto causa maior complexidade, e não era esperado que fosse feito dessa forma.